app/services/badges/award.rb

```diff
@@ -4,6 +4,8 @@ def self.call(user_relation, slug, message_markdown)
       return unless (badge_id = Badge.id_for_slug(slug))
 
       user_relation.find_each do |user|
+        next if user.banished?
```

Comment:
comment: `User#banished?` checks `username.starts_with?("spam_")`, i's already loaded in the fetched `user` object, this is as close to zero cost as we can get.